### PR TITLE
Fix flow gravity term in fractures

### PR DESCRIPTION
**Pull request by CusiniM**
The gravity term is currently not correctly accounted for in flow simulations involving fractures because the gravity coefficient is no properly computed for elements added by the SurfaceGenerator. This PR solves this issue and adds appropriate integratedTests.  

rebaseline PR:

https://github.com/GEOSX/integratedTests/pull/310

### CODE REVIEW

**Comment by CusiniM** [comment]
The fix seems to work for conforming Fractures:

Screen Shot 2023-03-17 at 2 17 14 PM

File: src/coreComponents/physicsSolvers/surfaceGeneration/SurfaceGenerator.cpp [hunk 1]
```diff
@@ -510,6 +510,22 @@ real64 SurfaceGenerator::solverStep( real64 const & time_n,
     {
       targetSet.insert( ei );
     } );
+
+    // Compute gravity coefficient for new elements so that gravity term is correctly computed
+    real64 const gravVector[3] = LVARRAY_TENSOROPS_INIT_LOCAL_3( gravityVector() );
+
+    if( fractureSubRegion.hasWrapper( fields::flow::gravityCoefficient::key()) )
```
Comment (francoishamon) [inline src/coreComponents/physicsSolvers/surfaceGeneration/SurfaceGenerator.cpp:517]:
```suggestion
    if( fractureSubRegion.hasField< fields::flow::gravityCoefficient >() )
```

File: src/coreComponents/physicsSolvers/surfaceGeneration/EmbeddedSurfaceGenerator.cpp [hunk 2]
```diff
@@ -277,6 +278,31 @@ real64 EmbeddedSurfaceGenerator::solverStep( real64 const & GEOSX_UNUSED_PARAM(
...
+  {
+    ElementRegionManager & elemManager = meshLevel.getElemManager();
+    SurfaceElementRegion & fractureRegion = elemManager.getRegion< SurfaceElementRegion >( this->m_fractureRegionName );
+
+    EmbeddedSurfaceSubRegion & fractureSubRegion = fractureRegion.getUniqueSubRegion< EmbeddedSurfaceSubRegion >();
+
+    // Compute gravity coefficient for new elements so that gravity term is correctly computed
+    real64 const gravVector[3] = LVARRAY_TENSOROPS_INIT_LOCAL_3( gravityVector() );
+
+    if( fractureSubRegion.hasWrapper( fields::flow::gravityCoefficient::key()) )
```
Comment (francoishamon) [inline src/coreComponents/physicsSolvers/surfaceGeneration/EmbeddedSurfaceGenerator.cpp:293]:
```suggestion
    if( fractureSubRegion.hasField< fields::flow::gravityCoefficient >() )
```

**Review by francoishamon — APPROVED** [review]
Looks great

**Pull request merged**